react-native/README.md

```diff
@@ -35,6 +37,66 @@ import * as PXBThemes from '@pxblue/react-native-themes';
 </ThemeProvider>
 ```
 
+### Dark Theme
+
+To use the dark theme in your application, simply wrap the app in a `Provider` and pass in your desired theme (`blueDark`).
+
+```tsx
+import { Provider as ThemeProvider} from 'react-native-paper';
+import * as PXBThemes from '@pxblue/react-native-themes';
+...
+<ThemeProvider theme={PXBThemes.blueDark}>
+    <App />
+</ThemeProvider>
+```
+
+#### Recommended Use With React Native Paper Components
+
+We provide an additional dark theme (`blueDarkAlt`) to be used with certain components in the React Native Paper library.
+The following components will need to have the `blueDarkAlt` theme applied:
+
+-   Appbar
+-   Avatar
+-   Badge
+-   Bottom Navigation
+-   Button (`contained` mode variant)
+-   FAB
+-   TextInput
+-   ProgressBar
+-   Snackbar
+
+One-Off Usage
+
+```tsx
+import { useTheme } from 'react-native-paper';
+import { blueDarkAlt } from '@pxblue/react-native-themes';
+const theme = useTheme();
+...
+<Badge size={24} visible theme={theme.dark ? blueDarkAlt : {}}></Badge>
+```
+
+Component Based Usage
+
+```tsx
+import { blueDarkAlt } from '@pxblue/react-native-themes';
+import React from 'react';
+import { Button, useTheme } from 'react-native-paper';
+...
+export const MyCustomButton: typeof Button = (props) => {
+    const theme = useTheme(props.theme);
+    return (
+        <Button
+            {...props}
+            theme={
+                props.mode === 'contained' && theme.dark
+                    ? Object.assign({}, JSON.parse(JSON.stringify(blueDarkAlt)), props.theme)
```

Comment:
A little technical question: why do we need to do this deep copy of `blueDarkAlt`? `blueDarkAlt` is not getting destructively modified here since the `target` parameter of your `Object.assign` is `{}`?

Reply:
Honestly, we might not need it in this situation. I needed it for another custom thing I did that did a little more manipulation of the object, I ended up just copy-pasting after that. I'll verify that we don't need it in this scenario and remove it if so.

Reply:
The reason is because both Object.assign and {...spreadOperator} do a shallow copy of nested data. So all of the stuff 1 level down in the object hierarchy gets passed by reference instead of copied. This leads to the problems that Tom was seeing where the original theme object was getting modified in memory and causing the colors to be janky.

We need to do a deep copy/clone. One way of doing that is stringify/parse...another would be to use the lodash deepClone function.

Reply:
I see, that makes more sense :) I didn't realize that object.assign does a shallow copy.